frontend/src/components/channels/HealthcheckAlert.tsx

```diff
@@ -0,0 +1,93 @@
+import { AlertTriangle, CheckCircle2 } from "lucide-react";
+import { Alert, AlertDescription, AlertTitle } from "src/components/ui/alert";
+import { ExternalLinkButton } from "src/components/ui/button";
+import { useHealthCheck } from "src/hooks/useHealthCheck";
+import { AlbyInfoIncident, HealthAlarm } from "src/types";
+
+type HealthCheckAlertProps = {
+  showOk?: boolean;
+};
+
+export function HealthCheckAlert({ showOk }: HealthCheckAlertProps) {
+  const { data: health } = useHealthCheck();
+
+  const ok = !health?.alarms?.length;
+
+  if (!health) {
+    return null;
+  }
+
+  if (!showOk && ok) {
+    return null;
+  }
+
+  function getAlarmTitle(alarm: HealthAlarm) {
+    // TODO: could show extra data from alarm.rawDetails
+    // for some alarm types
+    try {
+      switch (alarm.kind) {
+        case "alby_service":
+          return (
+            "Alby Services: " +
+            (alarm.rawDetails as AlbyInfoIncident[])
+              ?.map((incident) => `${incident.name} (${incident.status})`)
```

Comment:
this is all properly escaped?

Reply:
It is, but I tested it with fake data based on instatus' api for incidents. I'm not sure how to test this properly without us actually having an incident. For now I wrapped it in a try-catch just in case :scream:

Reply:
I was more thinking about code injections if somebody manages to make the API return some HTML/JS.

Reply:
Ah, React protects against this against this already.